tests/Cms/Pages/PageTest.php

```diff
@@ -1111,4 +1111,69 @@ public function testPageModel()
 
         Page::$models = [];
     }
+
+    public function testPageController()
+    {
+        $app = new App([
+            'roots' => [
+                'index' => __DIR__ . '/fixtures/PageTest'
+            ],
+            'site' => [
+                'children' => [
+                    [
+                        'slug' => 'test',
+                        'content' => [
+                            'title' => 'Test Title',
+                        ]
+                    ]
+                ]
+            ],
+            'controllers' => [
+                'default' => function ($page) {
+                    $page = $page->changeTitle('New Title');
+
+                    return ['page' => $page];
+                }
+            ]
+        ]);
+
+        $app->impersonate('kirby');
+        $page = $app->page('test');
+        $data = $page->controller();
+
+        $this->assertCount(4, $data);
+        $this->assertInstanceOf('Kirby\Cms\Page', $data['page']);
+        $this->assertSame('New Title', $data['page']->title()->value());
+    }
+
+    public function testPageControllerInvalid()
```

Comment:
```suggestion
    public function testControllerInvalid()
```

**Note:** Maybe this test could be combined with the `testController` by adding a second page and controller to that test and adding the test code to the end of that test. This would reduce the duplication of the boilerplate code.